scheduler/workflow.go

```diff
@@ -335,26 +336,38 @@ func (s *schedulerWorkflow) StateString() string {
 }
 
 func (s *schedulerWorkflow) workJobs(prs []*processNode, pus []*publishNode, t *task, pj job) {
+	// create a waitgroup track pending goroutines submitting jobs
+	wg := &sync.WaitGroup{}
 	for _, pr := range prs {
-		j := newProcessJob(pj, pr.Name(), pr.Version(), pr.InboundContentType, pr.config.Table(), t.metricsManager, t.id)
-		errors := t.manager.Work(j).Promise().Await()
-		if len(errors) != 0 {
-			t.failedRuns++
-			t.lastFailureTime = t.lastFireTime
-			t.lastFailureMessage = errors[len(errors)-1].Error()
-			return
-		}
+		go func() {
+			wg.Add(1)
+			defer wg.Done()
+			j := newProcessJob(pj, pr.Name(), pr.Version(), pr.InboundContentType, pr.config.Table(), t.metricsManager, t.id)
+			errors := t.manager.Work(j).Promise().Await()
```

Comment:
My only thing about that is the WaitGroup and go outine use here is pretty idiomatic. The promise and queueing is not go channels so it is unique. I feel like this goroutine use is safer to understand and support longer term since it is a pretty normal use.

I did look at AndThen and I feel its use here would be harder to understand while note necessarily better.